compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1656,7 +1655,19 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     self.tcx.sess.emit_err(AsyncCoroutinesNotSupported { span }),
                 );
             }
-            Some(hir::CoroutineKind::Coroutine) | None => {
+            Some(hir::CoroutineKind::Coroutine(_)) => {
+                if !self.tcx.features().coroutines {
+                    rustc_session::parse::feature_err(
+                        &self.tcx.sess.parse_sess,
+                        sym::coroutines,
+                        span,
+                        "yield syntax is experimental",
```

Comment:
Huh, weird that this wasn't here earlier...

Reply:
It was there! I just split out the branches so the code got duplicated.

Reply:
Oh, there it is, down below. Duh!